packages/engine/Source/Core/ITwin.js

```diff
@@ -0,0 +1,306 @@
+import defined from "./defined.js";
+import DeveloperError from "./DeveloperError.js";
+import Resource from "./Resource.js";
+import RuntimeError from "./RuntimeError.js";
+
+/**
+ * @enum {string}
+ */
+const ExportStatus = Object.freeze({
+  NotStarted: "NotStarted",
+  InProgress: "InProgress",
+  Complete: "Complete",
+  Invalid: "Invalid",
+});
+
+/**
+ * Type of an export currently, only GLTF and 3DFT are documented
+ * The CESIUM option is what we were told to use with Sandcastle
+ * I've also seen the IMODEL one but don't know where it's from
+ * @enum {string}
+ */
+const ExportType = Object.freeze({
+  "3DFT": "3DFT",
+  GLFT: "GLTF",
+  IMODEL: "IMODEL",
+  CESIUM: "CESIUM",
+});
+
+/**
+ * @typedef {Object} GeometryOptions
+ * @property {boolean} includeLines
+ * @property {number} chordTol
+ * @property {number} angleTol
+ * @property {number} decimationTol
+ * @property {number} maxEdgeLength
+ * @property {number} minBRepFeatureSize
+ * @property {number} minLineStyleComponentSize
+ */
+
+/**
+ * @typedef {Object} ViewDefinitionFilter
+ * @property {string[]} models Array of included model IDs.
+ * @property {string[]} categories Array of included category IDs.
+ * @property {string[]} neverDrawn Array of element IDs to filter out.
+ */
+
+/**
+ * @typedef {Object} StartExport
+ * @property {string} iModelId
+ * @property {string} changesetId
+ * @property {ExportType} exportType Type of mesh to create. Currently, only GLTF and 3DFT are supported and undocumented CESIUM option
+ * @property {GeometryOptions} geometryOptions
+ * @property {ViewDefinitionFilter} viewDefinitionFilter
+ */
+
+/**
+ * @typedef {Object} Link
+ * @property {string} href
+ */
+
+/**
+ * @typedef {Object} Export
+ * @property {string} id
+ * @property {string} displayName
+ * @property {ExportStatus} status
+ * @property {StartExport} request
+ * @property {{mesh: Link}} _links
+ */
+
+/**
+ * @typedef {Object} ExportResponse
+ * @property {Export} export
+ */
+
+/**
+ * Default settings for accessing the iTwin platform.
+ *
+ * Keys can be created using the iModels share routes {@link https://developer.bentley.com/apis/imodels-v2/operations/create-imodel-share/}
```

Comment:
Please note, this API has the following note:
> Important : Share operations are in closed preview mode currently. Hence only selected applications can utilize the Share API.

fyi @robertas-kerpys, we will need to work with Cesium team here

Reply:
I don't think we plan to use the model share api here at all, I don't think it gives us what we need. This was an early comment when I thought we would be.

We definitely will need a good strategy for auth in general though. it's one of the biggest open questions we have right now. I think that discussion should live outside this specific thread.

Reply:
For demo purposes, we're working right now on adding shared keys to the mesh export service. I expect to have it deployed in the next 4 weeks, if all goes as planned.

Reply:
Hi @danieliborra! We're targeting these changes for the end of November of so that we can (ideally) ship them with the December 1 CesiumJS release. Given that, what is the recommended approach?